src/cryptography/hazmat/bindings/openssl/binding.py

```diff
@@ -74,6 +79,17 @@ def _register_osrandom_engine(cls):
     def _ensure_ffi_initialized(cls):
         with cls._init_lock:
             if not cls._lib_loaded:
+                conditional_lib = _ConditionalLibrary()
+                for attr in dir(lib):
```

Comment:
All of this should probably live in the __init__ for  ``_ConditionalLibrary`` so that the API ends up being something like:

```python
class Binding:
    def _ensure_ffi_initialized(cls):
        ...
        cls.lib = _ConditionalLibrary(lib, CONDITIONAL_NAMES)
```